module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,63 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+#include <algorithm>
+#include <map>
+#include <iostream>
+using EC = ErrorCode;
+
+std::string getErrorMessage(const ErrorCode &err)
+{
+    std::string message;
+    std::map<ErrorCode, std::string> errorMap = {
+        {EC::Ok, "YourPasswordIsOK"},
+        {EC::PasswordNeedsAtLeastNineCharacters, "NeedsAtLeastNineCharacters"},
+        {EC::PasswordNeedsAtLeastOneNumber, "NeedsAtLeastOneNumber"},
+        {EC::PasswordNeedsAtLeastOneSpecialCharacter, "NeedsOneSpecialCharacter"},
+        {EC::PasswordNeedsAtLeastOneUppercaseLetter, "NeedsAtLeastOneUppercaseLetter"},
+        {EC::PasswordsDoesNotMatch, "PasswordsDoesNotMatch"}};
+
+    if (errorMap.find(err) != errorMap.end())
+    {
+        message = errorMap[err];
+    }
+    return message;
+}
+
+bool doesPasswordsMatch(const std::string &passwd1,
+                        const std::string &passwd2)
```

Comment:
Connecting `&` with a type name makes it less possible to misunderstand that the `&` is a part of variables name.
```suggestion
bool doesPasswordsMatch(const std::string& passwd1,
                        const std::string& passwd2)
```